bin/script.js

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env node
+
+const yargs = require('yargs');
+const fs = require('fs');
+const yaml = require('js-yaml');
+
+const options = yargs.usage('Usage: --vendor <file> [--vendor-id <id>]').option('v', {
+  alias: 'vendor',
+  describe: 'Path to vendor index file',
+  type: 'string',
+  demandOption: true,
+  default: './vendor/index.yaml',
+}).argv;
+
+function requireFile(path) {
+  if (path.toLowerCase() !== path) {
+    return Promise.reject(new Error(`${path} is not lowercase`));
+  }
+  return new Promise((resolve, reject) => {
+    fs.stat(path, (err) => {
+      if (err) {
+        reject(new Error(`stat ${path}: ${err.code}`));
+      } else {
+        resolve();
+      }
+    });
+  });
+}
+
+const vendorProfiles = {};
+
+const vendors = yaml.load(fs.readFileSync(options.vendor));
+
+vendors.vendors.forEach((v) => {
+  const key = v.id;
+  const folder = `./vendor/${v.id}`;
+  const vendorIndexPath = `${folder}/index.yaml`;
+  fs.stat(vendorIndexPath, (err) => {
+    if (err) {
+      if (err.code !== 'ENOENT') {
+        console.error(`${key}: index file: ${err.code}`);
+        process.exit(1);
+      }
+      return;
+    }
+
+    const vendor = yaml.load(fs.readFileSync(vendorIndexPath));
+
+    const codecs = {};
+
+    vendor.endDevices.forEach(async (d) => {
+      const deviceFilePath = `${folder}/${d}.yaml`;
+      const endDevice = yaml.load(fs.readFileSync(deviceFilePath));
+
+      // Assuming firmwareVersion and hardwareVersion are correctly extracted
+      // Iterate through firmware versions to access profiles
+      endDevice.firmwareVersions.forEach((version) => {
+        Object.keys(version.profiles).forEach(async (region) => {
+          const profilePath = `./vendor/${v.id}/${version.profiles[region].id}.yaml`;
+          const profile = yaml.load(fs.readFileSync(profilePath));
+
+          // Use '0' as a fallback but decide not to proceed if it is '0'
+          let profileID = profile.vendorProfileID || '0';
```

Comment:
You should convert this to a number (and not treat it as a string). This is because `'0002'` doesn't equal `'2'` while 2 equals 2.